src/classes/RD2_DataMigrationBase_BATCH.cls

```diff
@@ -152,6 +152,7 @@ public abstract class RD2_DataMigrationBase_BATCH implements Database.Batchable<
         //enable enhanced Recurring Donations during the batch transaction
         if (!RD2_EnablementService.isRecurringDonations2Enabled && isDryRunMode) {
             RD2_EnablementService.isRecurringDonations2EnabledOverride = true;
+            RD2_ValidationService.isRD2MetadataDeployed = true;
```

Comment:
The validation service domain is not the enablement, thus, based on the recommended changes in the enablement service below, this can be replaced with `RD2_EnablementService.isMetaDataDeployed = true;`.
Btw, nice catch that validation needs to succeed before actual enablement occurs.